### PR TITLE
Add more monitors in ops

### PR DESCRIPTION
**Pull request by JanisGailis**
What the title says.

### CODE REVIEW

**Comment by JanisGailis** [comment]
@mzuehlke You're the expert on observing monitors! Can you please check this out? I felt a bit unsure in places if what I'm doing is the correct thing to do. Especially in correlation (monitor.observing encompasses a lot of code) and in index.py and timeseries.py.

**Comment by mzuehlke** [comment]
I only looked at the operators you had modified. And improved the code in some places.
`_pearsonr` I looked a bit more closely by making sure the input is chunked and the then look at the output of the `RecordingMonitor` and by locally enabling `_DEBUG_DASK_PROGRESS` in "monitor.py".

File: cate/ops/correlation.py [hunk 1]
```diff
@@ -251,8 +253,14 @@ def _pearsonr(x: xr.DataArray, y: xr.DataArray, monitor: Monitor) -> xr.Dataset:
         # deferred processing.
         # Comparing with NaN produces warnings that can be safely ignored
         default_warning_settings = np.seterr(invalid='ignore')
-        r.values[r.values < -1.0] = -1.0
-        r.values[r.values > 1.0] = 1.0
+        with monitor.child(1).observing("task 1"):
+            negativ_r = r.values < -1.0
```
Comment (JanisGailis) [inline cate/ops/correlation.py]: @mzuehlke 
So, we only monitor when we try to access the values? As in, processing from lines 239-247 would be deferred until then?

**Comment by mzuehlke** [comment]
I will in a calm moment try to make the dask progress monitor a bit smarter. Because I find this multiple observing calls not the nicest solution.....

**Comment by JanisGailis** [comment]
So, for future reference - as it is now, we can only wrap a single line in ``with monitor.observing():`` right?

**Comment by mzuehlke** [comment]
Currently: yes

**Pull request merged**